lib/karafka/server.rb

```diff
@@ -67,7 +67,12 @@ def start_supervised
       # Stops Karafka with a supervision (as long as there is a shutdown timeout)
       # If consumers won't stop in a given timeframe, it will force them to exit
       def stop_supervised
-        Karafka.monitor.instrument('server.stop', {})
+        # Due to the fact that this is called in the trap context, there is a chance that
+        # instrumentation listeners contain things that aren't allowed from within a trap
+        # context. To bypass that (instead of telling users not to do things they need to)
+        # we just spin up a thread ti instrument server.stop and server.stop.error and wait
+        # until they're finished
```

Comment:
Because this is called in the trap context, there is a chance that instrumentation listeners contain things that aren't allowed from within a trap context. To bypass that (instead of telling users not to do things they need to) we spin up a thread to instrument server.stop and server.stop.error and wait until they're finished